Manual.md

```diff
@@ -0,0 +1,145 @@
+
+***
+
+## Introduction
+
+Each file under `packages/` is a recipe for `crew`, each recipe contains the instructions to compile from source and/or install as a binary package using `crew`. These package files are called `.rb`'s within this repository, but are otherwise know as ruby scripts.  
+
+The `.rbs`'s are ruby scripts called by `crew`<sup>(The Chromebrew package manager)</sup>. Each `.rb` must follow a explicit set of rules laid out by `crew`, these rules are provided within this Manual.
+
+***
+# Setup
+`crew` uses the following layout to install/build packages.
+
+1. `fetch` - During the fetch phase `crew` uses `curl --ssl`  to fetch the package
+2. `extract` - Once the the package has been fetched, `crew` extracts the package
+3. `self.prebuild` - If provided, only `sed` commands should be used during this stage
+	- During this stage `crew` `cd`'s into the extracted directory, and stays there until the `preinstall` stage 
+4. `self.patch` - Only `patch` commands should be used during this stage	- If provided
+5. `self.build` - Package is compiled
+6. `self.check` - Package is then checked - If provided
+7. `self.preinstall` - Pre-install checks/edits are done - If provided 
+	- `crew` now `cd`'s out of the directory provided by the package
+8. `self.install` - Commands to install to `#{CREW_DEST_DIR}` are run
+	- Only sourced when the package is being built
+		- `self.install` is not required to be defined
+			- Install may be carried out during `self.build` or `self.preinstall`
```

Comment:
I would dare mention to say this is the ONLY required function in this entire section.  We definitely don't want to install the package in other functions because that could lead to confusion.

Reply:
I had stated it as such as some `nodejs` packages install and compile within the same script, I'll change it for your suggestion though.